src/LinearRemapSE0.cpp

```diff
@@ -849,26 +846,41 @@ void LinearRemapSE4(
 		// Find the local remap coefficients
 		for (int j = 0; j < nOverlapFaces; j++) {
 			const Face & faceOverlap = meshOverlap.faces[ixOverlap + j];
-
-			int nOverlapTriangles = faceOverlap.edges.size() - 2;
-
-			// Loop over all sub-triangles of this Overlap Face
-			for (int k = 0; k < nOverlapTriangles; k++) {
-
-				// Cornerpoints of triangle
-				const Node & node0 = nodesOverlap[faceOverlap[0]];
-				const Node & node1 = nodesOverlap[faceOverlap[k+1]];
-				const Node & node2 = nodesOverlap[faceOverlap[k+2]];
-
-				// Calculate the area of the modified Face
-				Face faceTri(3);
-				faceTri.SetNode(0, faceOverlap[0]);
-				faceTri.SetNode(1, faceOverlap[k+1]);
-				faceTri.SetNode(2, faceOverlap[k+2]);
-
-				double dTriangleArea =
-					CalculateFaceArea(faceTri, nodesOverlap);
-
+			int nbEdges = faceOverlap.edges.size();
+			int nOverlapTriangles = 1;
+			Node center; // not used if nbEdges == 3
+            if (nbEdges > 3) { // decompose from center in this case
+                nOverlapTriangles = nbEdges;
+                for (int k = 0; k < nbEdges; k++) {
+                    const Node &node = nodesOverlap[faceOverlap[k]];
+                    center = center + node;
+                }
+                center = center / nbEdges;
+                double magni = sqrt(
+                        center.x * center.x + center.y * center.y
+                                + center.z * center.z);
+                center = center / magni; // project back on sphere of radius 1
+            }
+
+            Node node0, node1, node2;
+            double dTriangleArea;
+
+            // Loop over all sub-triangles of this Overlap Face
+            for (int k = 0; k < nOverlapTriangles; k++) {
+                if (nbEdges == 3) // will come here only once, nOverlapTriangles == 1 in this case
+                {
+                    node0 = nodesOverlap[faceOverlap[0]];
```

Comment:
Same comments as above. Can refactor decomposition code and make a single call rather than repeating this for every discretization combo